### ISSUE TITLE
odo support for quarkus

### ISSUE THREAD

**Issue by adisky** [openshift/odo#2800]
/kind user-story

## User Story
Quarkus is a full-stack, Kubernetes-native Java framework, As a user i want run applications based on Quarkus with odo.  

## Acceptance Criteria 
- [x] odo workflow should work well with quarkus apps
`odo create` `odo push`  `odo delete` `odo url create`, accessing application from given url.
- [x] `odo catalog list components` should list down quarkus as supported app
- [x]  Example Devfile needs to be added in devfile registry.
https://github.com/elsony/devfile-registry/tree/master/devfiles
https://github.com/eclipse/che-devfile-registry/blob/master/devfiles/quarkus/devfile.yaml
- [ ] Integration tests for quarkus support in odo
- [x] Update the relevant Documents 

## Links 
- Feature Request: #1444 
- Related issue: 

/kind user-story

**Comment by adisky** [comment]
Dependent PR's
https://github.com/openshift/odo/pull/2763
https://github.com/openshift/odo/pull/2735
https://github.com/openshift/odo/pull/2762

**Comment by girishramnani** [comment]
Quarkus has a nice remote development mode, do we wanna verify/implement that?

**Comment by adisky** [comment]
> Quarkus has a nice remote development mode, do we wanna verify/implement that?

Thanks for letting me now about this, yes I can implement that but should we make it part of this user story(i considered basic cases here) or log a separate issue??

**Comment by adisky** [comment]
Devfile under review https://github.com/elsony/devfile-registry/pull/11
Also Depends on https://github.com/openshift/odo/issues/2820

**Comment by adisky** [comment]
for quarkus dev mode, we need not run `devRun: mvn quarkus:dev` command with every odo push, and we just need to run `devBuild: mvn compile` only for first build,  so we can use `devInit` here.
```
---
apiVersion: 1.0.0
metadata:
  generateName: quarkus-
components:
  - type: dockerimage
    alias: tools
    image: quay.io/eclipse/che-quarkus:7.11.0
    memoryLimit: 1512Mi
    mountSources: true
    volumes:
      - name: m2
        containerPath: /home/user/.m2
    endpoints:
      - name: '8080/tcp'
        port: 8080
commands:
  - name: devInit
    actions:
      - type: exec
        component: tools
        command: "mvn compile"
  - name: devRun
    attributes:
        restart: "false"
    actions:
      - type: exec
        component: tools
        command: "mvn quarkus:dev"
```

**Comment by adisky** [comment]
Quarkus support has been added in odo,  quarkus devfile is now available at https://github.com/elsony/devfile-registry/tree/master/devfiles/quarkus, 

Following things remaining
- Integration tests for quarkus support in odo
- Update the relevant Documents

**Comment by boczkowska** [comment]
@adisky would it be a good idea to reach out to Quarkus docs team to mention odo/create a small user-story in their docs too?
They do have some docs on how to deploy Quarkus on OCP: https://access.redhat.com/documentation/en-us/red_hat_build_of_quarkus/1.3/html-single/deploying_quarkus_applications_on_red_hat_openshift_container_platform/index

**Comment by adisky** [comment]
> @adisky would it be a good idea to reach out to Quarkus team to mention odo/create a small user-story in their docs too?

I am not sure about this, as we are supporting it under `experimental` mode, @kadel @girishramnani should we do this now?

> Both upstream and downstream have some docs on how to deploy Quarkus on OCP

Yup I would add that.

**Comment by adisky** [comment]
for devfile v2, it is dependent on https://github.com/openshift/odo/issues/2936

**Comment by dharmit** [comment]
Blocked on #2936

**Comment by adisky** [comment]
I have tested quarkus with application https://github.com/quarkusio/quarkus-quickstarts/tree/master/getting-started and https://github.com/adisky/hello-quarkus, i could add any of them as a starter project.

**Comment by girishramnani** [comment]
as of now - we dont need special integration tests for quarkus as we have tests in the devfile registry for these
